rma/models/rma.py

```diff
@@ -670,21 +661,94 @@ def action_rma_send(self):
         }
 
     def _add_message_subscribe_partner(self):
+        self.ensure_one()
         if self.partner_id and self.partner_id not in self.message_partner_ids:
             self.message_subscribe([self.partner_id.id])
 
+    def _product_is_storable(self, product=None):
+        product = product or self.product_id
+        return product.type in ["product", "consu"]
+
+    def _prepare_procurement_group_values(self):
+        return {
+            "move_type": "direct",
+            "partner_id": self and self.partner_shipping_id.id or False,
+            "name": self and ", ".join(self.mapped("name")) or False,
+        }
+
+    def _prepare_procurement_values(
+        self, warehouse=None, scheduled_date=None, group=None
+    ):
+        self.ensure_one()
+        group = group or self.procurement_group_id
+        if not group:
+            group = self.env["procurement.group"].create(
+                self._prepare_procurement_group_values()
+            )
+        return {
+            "company_id": self.company_id,
+            "group_id": group,
+            "date_planned": scheduled_date or fields.Datetime.now(),
+            "warehouse_id": warehouse or self.warehouse_id,
+            "partner_id": group.partner_id.id,
+            "priority": self.priority,
+        }
+
+    def _prepare_reception_procurement_values(self, group=None):
+        values = self._prepare_procurement_values(group=group)
+        values.update(
+            {
+                "rma_receiver_ids": [(6, 0, self.ids)],
+            }
+        )
```

Comment:
Better to not use `update` as it's uglier:

```suggestion
        vals["rma_receiver_ids"] = Command.set(self.ids)
```